server/public/model/utils.go

```diff
@@ -732,6 +732,11 @@ func IsValidHTTPURL(rawURL string) bool {
 	return true
 }
 
+func IsValidURI(rawURI string) error {
+	_, err := url.ParseRequestURI(rawURI)
```

Comment:
`url.ParseRequestURI` is not well-suited for use as a general-purpose URI validator. Particularly it fails to parse URLs like `http://example.com#` because it assumes an incoming request will never contain the `#` character. I know this was already broken before, but while we're fixing things, we should fix them properly.

Here's a more reasonable URI validator that I just banged out, maybe we could do something along these lines: https://go.dev/play/p/QOMLG-26a9s